### PR TITLE
Refactor external sst file ingestion job

### PR DESCRIPTION
**Pull request by jowlyzhang**
Updates some documentations and invariant assertions after #12257 and #12284. Also refactored some duplicate code and improved some error message and preconditions for errors.

Test plan:
Existing unit tests

### CODE REVIEW

**Review by hx235 — APPROVED** [review]
LGTM

File: db/external_sst_file_ingestion_job.cc [hunk 1]
```diff
@@ -316,21 +317,9 @@ Status ExternalSstFileIngestionJob::Prepare(
...
-        continue;
-      }
-      Status s = fs_->DeleteFile(f.internal_file_path, io_opts, nullptr);
-      if (!s.ok()) {
-        ROCKS_LOG_WARN(db_options_.info_log,
-                       "AddFile() clean up for file %s failed : %s",
-                       f.internal_file_path.c_str(), s.ToString().c_str());
-      }
-    }
+    DeleteInternalFiles();
```
Comment (cbi42) [inline db/external_sst_file_ingestion_job.cc:322]: It seems we call Cleanup() after Prepare() return non-ok status: https://github.com/facebook/rocksdb/blob/6e88126dd33219727021d9d9f84b42a47cc647b1/db/db_impl/db_impl.cc#L5850

Comment (jowlyzhang) [reply db/external_sst_file_ingestion_job.cc:322]: Yes, do you mean that we should remove this `DeleteInternalFiles` call from `Prepare`, so we don't get the "clean up for file failed" for this scenario?

Comment (cbi42) [reply db/external_sst_file_ingestion_job.cc:322]: Yeah, I did not think about the warnings but just that we don't need to call `DeleteInternalFiles()` here.

**Pull request closed without merging**